src/mococrw/openssl_wrap.h

```diff
@@ -1067,6 +1056,64 @@ void _EVP_PKEY_verify(EVP_PKEY_CTX *ctx,
                       const unsigned char *sig, size_t siglen,
                       const unsigned char *tbs, size_t tbslen);
 
+/**
+ * Initialize the MD_CTX for signing, using a given digest-type and a given
+ * public key.
+ *
+ * @param ctx The context to be set up for signing.
+ * @param md The digest type (@see DigestTypes)
+ * @param pkey The public key to be used.
+ *
+ * @throw OpenSSLException if an error occurs in the underlying OpenSSL function.
+ */
+void _EVP_DigestSignInit(EVP_MD_CTX* ctx, DigestTypes md, EVP_PKEY* pkey);
+
+/**
+ * Perform a digital signature of the message tbs and stores the signature in sigret.
+ *
+ * @param ctx The context used for signing.
+ * @param sigret The buffer to write the signature to.
+ * @param siglen The length of the generated signature.
+ * @param tbs Message to be signed.
+ * @param tbslen Length of the message to be signed.
+ *
+ * @throw OpenSSLException if an error occurs in the underlying OpenSSL function.
+ */
+void _EVP_DigestSign(EVP_MD_CTX* ctx,
+                     unsigned char* sigret,
+                     size_t* siglen,
+                     const unsigned char* tbs,
+                     size_t tbslen);
+
+/**
+ * Initialize the MD_CTX for signing, using a given digest-type and a given
+ * public key.
+ *
+ * @param ctx The context to be set up for signing.
+ * @param md The digest type (@see DigestTypes)
+ * @param pkey The public key to be used.
+ *
+ * @throw OpenSSLException if an error occurs in the underlying OpenSSL function.
+ */
+void _EVP_DigestVerifyInit(EVP_MD_CTX* ctx, DigestTypes type, EVP_PKEY* pkey);
+
+/**
+ * Verifies a given digital signature of a message tbs.
+ *
+ * @param ctx The context used for signing.
+ * @param sigret The signature to be verified.
+ * @param siglen The length of the the signature to be verified.
+ * @param tbs Signed message.
+ * @param tbslen Length of the signed message.
+ *
+ * @throw OpenSSLException if an error occurs in the underlying OpenSSL function.
+ */
+void _EVP_DigestVerify(EVP_MD_CTX *ctx,
```

Comment:
How is the information whether the signature was valid or not transported to the caller?

Reply:
An OpenSSLException is thrown. (I added that to the comment)